src/gui/sdlmain.cpp

```diff
@@ -1109,7 +1109,15 @@ static void setup_presentation_mode(FrameMode &previous_mode)
 		                  "fullscreen, nor vsync'd, nor < 140Hz");
 #endif
 
-		if (supported_rate >= dos_rate) {
+		// Text-based programs (including DOSBox itself) often re-draw
+		// text elements in VGA memory immediately on-demand without
+		// waiting for vblank. This means that throttled VFR's follow up
+		// frame has often already been modified (or is new) So for text
+		// modes, we avoid throttling.
+		//
+		const auto in_text_mode = VGA_InTextMode();
+
+		if (supported_rate >= dos_rate || in_text_mode) {
```

Comment:
Please let's not special case text modes in any way; we've discussed this before a few times and ultimately you agreed with me that video output is video output 😄 So let's not bring any text-mode specific special handling back.

An application that uses a graphics mode and updates the screen sporadically without waiting for vblank can manifest the _exact same issue_—there is zero difference, it's 100% immaterial whether we're in a text mode or a graphics mode, the "update patterns" matter only. Image viewers fall into this category, but also the in-game menu of many games where the screen is completely static (if they don't do vsync).

Reply:
Ok, I'll check how that looks tomorrow. Maybe the other change is sufficient.

Reply:
Well, just forget about differentiating text vs graphics modes altogether. You're thinking how the problem can manifest is 100% correct; the error is only when assuming this can only happen in text modes. No, it can happen in _any_ video mode!

Reply:
Removed!👍